### DISCUSSION TITLE
Looking for Contributors to Help Improve Bloxtor

### DISCUSSION THREAD

**Discussion by a19836** [community/community#141537] [Programming Help]
Hello everyone,

I'm new to GitHub and have just added a new project called **Bloxtor**. Bloxtor is a professional and intelligent framework for building web apps, designed for No-Coders, Low-Coders, and Coders alike. It serves as a powerful code parser and converter, seamlessly transforming code into visual environments, and vice versa. Plus, it’s completely **FREE** and **OPEN SOURCE**.

I’m reaching out to see if anyone would be interested in helping improve this framework and sharing it with the world.

If you'd like to contribute, please check out the project here: [Bloxtor on GitHub](https://github.com/a19836/bloxtor).

**Note**: If you’re not able to contribute code but would still like to help, we’d love your feedback as well! Any thoughts or suggestions are greatly appreciated and can help us make Bloxtor even better.

**What do you think about the Bloxtor framework?** We’d love to hear your thoughts, suggestions, or feedback!

Thank you in advance for your support!

**Comment by tpmst** [comment]
Hello @a19836,

I noticed you're interested in integrating AI into your application. I'd be happy to help! Here are a few options we could consider:

 - Developing a custom AI model that runs alongside your application.
 - Utilizing established models like GPT-4 or GPT-3.5 Turbo.
 - Integrating GitHub Copilot for coding assistance.

If we opt for an API from a major provider with a robust coding LLM, who would handle the associated costs? Alternatively, if you're leaning towards training your own model, where do you plan to conduct this training?

Additionally, if you're considering integrating ads to sustain the API costs, which service would you prefer to use, such as Google AdMob or another provider?

**Reply by a19836** [reply]
Hello Tom,

Thank you for reaching out!

We appreciate your suggestions for integrating AI into Bloxtor. The ideal solution will depend on which specific features we aim to enhance. Based on the [TODO.md](https://github.com/a19836/bloxtor/blob/main/TODOS.md), we've initial identified four key areas where AI could assist Bloxtor developers:

1. **Automatically generating HTML layouts** when creating or designing pages, templates, or views in the Visual and Advanced Editors.
2. **Automatically generating PHP code** when creating new services in the Business Logic Editor.
3. **Automatically generating SQL statements** when creating queries in the SQL Editor.

For these features, we envision a new menu option that opens a popup where developers can describe what they need in natural language, and the framework will generate the relevant code—HTML, PHP, or SQL—directly into the appropriate section.

Once these add-ons are in place, we could move on to the fourth idea:

4. **Creating a project from scratch using natural language**. In the "Create Project" popup, we could add a new tab where developers describe their project requirements in natural language, and Bloxtor would automatically generate the necessary code—database structures, SQL queries, business logic services, and interface elements—resulting in a fully functional app ready for the end user. This feature would be complex, and we should discuss the details further as we approach its implementation.

For these use cases, we believe GPT-4 or GPT-3.5 would be the most suitable technologies, though some customization and training would likely be necessary. However, **we are not experts in AI and are open to other suggestions and third-party services...**

Regarding API costs, this could be a premium feature, where developers cover the associated costs. We could implement an activation button where they provide payment details. We haven’t finalized the specifics of this payment process yet, but we also think that this part is taken care directly from GPT and we don't need to worry about it??? 
Additionally, offering a limited free version might be interesting, but we’re unsure if this would be possible with GPT.

As for cost-sharing via ads, such as Google AdMob, we prefer not to implement this at the moment. **Our goal is to maintain Bloxtor as a clean and distraction-free environment where developers can focus comfortably. Introducing ads might add unnecessary noise and reduce usability.**

Regarding the custom AI model, we don’t currently have the technical expertise to make an informed decision. We're uncertain about the training complexity and server resources required. **Since Bloxtor is a lightweight framework** (and we would like to keep it this way), we initially planned to keep AI processing external, allowing developers to activate it as an optional service. But we are open to suggestions...

Regarding the GitHub Copilot integration for coding assistance, it could also be interesting for the Business Logic Editor. However, we think it could be done separately in addition to GPT. We had also considered integrating Copilot into the Business Logic Editor before, but we’re not sure if it makes sense to add it together with GPT. What do you think?

*PS: Once you’re ready to start contributing, please create a new branch so we can continue working on other features in parallel.*

Looking forward to your thoughts and next steps!

Best regards

**Reply by a19836** [reply]
Hello Tom,

In order to contribute to this project, please fork our [Bloxtor repo](https://github.com/a19836/bloxtor.git) and follow the steps in the following [tutorial](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project) to better understand how can you do it.

Cheers

**Reply by a19836** [reply]
Hello Tom,

We've added you as a Bloxtor contributor. You should have received an email.
Please read the [CONTRIBUTING.md](https://github.com/a19836/bloxtor/blob/main/CONTRIBUTING.md) file before contributing.
Thanks for your help...

Cheers ;)
